src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -357,11 +367,27 @@ private async Task<IEnumerable<PackageReference>> GetPackageReferencesAsync(Pack
                 _lastTimeAssetsModified = fileInfo.LastWriteTimeUtc;
             }
 
-            return packageSpec
+            // get the installed packages
+            var installedPackages = packageSpec
                .TargetFrameworks
                .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName, _installedPackages, assetsPackageSpec, targets))
                .GroupBy(p => p.PackageIdentity)
                .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First());
+
+            // if we're reading the lock file, also update _transitivePackages
+            if (targets != null)
+            {
+                _transitivePackages = new Dictionary<string, PackageReference>(StringComparer.OrdinalIgnoreCase);
+                var _transitivePackages4 = targets
```

Comment:
naming: Probably don't need the 4 there :)

Reply:
oops! Fixed.